edx/frontend-app-discussions
# [WIP] [BD-03] [TNL-7353] Prototype: Create React-based forum post editor

In this PR I replaced textarea in add post component with WYSIWYG editor based on https://github.com/jpuri/react-draft-wysiwyg which in turn based on https://draftjs.org/.

Relevant commit: https://github.com/edx/frontend-app-discussions/commit/ac5b9968b42c218009f53080afc1cf1f40c22d17

**JIRA tickets**:
- [TNL-7353](https://openedx.atlassian.net/browse/TNL-7353)

**Dependencies**:

This PR depends on a multiple other PRs which depends on each other:
https://github.com/edx/frontend-app-discussions/pull/6 -> https://github.com/edx/frontend-app-discussions/pull/2 -> https://github.com/edx/frontend-app-discussions/pull/1

**Screenshots**:
![Screenshot from 2020-10-13 14-10-46](https://user-images.githubusercontent.com/18251194/95865876-d5892d00-0d6f-11eb-9824-74065da3504a.png)


**Sandbox URL**: TBD

**Merge deadline**: None

**Testing instructions**:

1. Run LMS. It should have http://localhost:18000 address.
2. Clone https://github.com/open-craft/frontend-app-discussions repo, checkout to `0x29a/bb2970/post_editor` branch, install dependencies with `npm install`, run development server with `npm start`.
3. Go to http://localhost:2002/discussions/course-v1:edX+DemoX+Demo_Course/posts/mine.
4. Click `Add Post`.
5. Play with `Your question or idea` input.

**Author notes and concerns**:

1. I decided to use https://github.com/jpuri/react-draft-wysiwyg because implementing proper WYSIWYG editor from scratch on top of draft-js is for sure beyond the scope of this ticket.
2. If we want to show raw version of editor like on Github, we can use https://github.com/kadikraman/draftjs-md-converter library. We can also use it to get markdown representation of editor state, but there can be other alternatives.
3. Since this is a prototype, and we didn't implement all necessary thunks, it doesn't have features like uploading images. Also, it doesn't contain any styling.

**Reviewers**
